python/hail/expr/functions.py

```diff
@@ -640,6 +658,23 @@ def parse_locus(s, reference_genome=None):
     return construct_expr(ApplyMethod('Locus({})'.format(reference_genome.name), s._ast), TLocus(reference_genome),
                           s._indices, s._aggregations, s._joins, s._refs)
 
+@typecheck(pl=ArrayNumericExpression)
+def pl_dosage(pl):
```

Comment:
Should this allow a python list of length 3 as well?

Reply:
this should be `expr_array` and the element type should be checked in the method